app/models/booking.rb

```diff
@@ -0,0 +1,7 @@
+class Booking < ApplicationRecord
+  validates :check_in, presence: true, uniqueness: true
+  validates :check_out, presence: true, uniqueness: true
```

Comment:
I think if you do "uniqueness: true" in the booking validation, it will actually prevent any booking (regardless of space) of having the same date.
```suggestion
  validates :check_out, presence: true
```